// @cerbos/core/lib/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationFailed = exports.NotOK = exports.Status = void 0;
/**
 * Status codes returned by the Cerbos policy decision point server.
 *
 * @public
 */
var Status;
(function (Status) {
    /**
     * The operation was cancelled.
     */
    Status[Status["CANCELLED"] = 1] = "CANCELLED";
    /**
     * The operation timed out.
     */
    Status[Status["DEADLINE_EXCEEDED"] = 4] = "DEADLINE_EXCEEDED";
    /**
     * The operation failed due to an internal error.
     */
    Status[Status["INTERNAL"] = 13] = "INTERNAL";
    /**
     * The operation was rejected because an argument was invalid.
     */
    Status[Status["INVALID_ARGUMENT"] = 3] = "INVALID_ARGUMENT";
    /**
     * The operation failed because a resource has been exhausted.
     */
    Status[Status["RESOURCE_EXHAUSTED"] = 8] = "RESOURCE_EXHAUSTED";
    /**
     * The operation was rejected because it did not have valid authentication credentials.
     */
    Status[Status["UNAUTHENTICATED"] = 16] = "UNAUTHENTICATED";
    /**
     * The operation was rejected because it did not have valid authentication credentials.
     */
    Status[Status["UNAVAILABLE"] = 14] = "UNAVAILABLE";
    /**
     * The operation is not supported.
     */
    Status[Status["UNIMPLEMENTED"] = 12] = "UNIMPLEMENTED";
    /**
     * An unknown error occurred.
     */
    Status[Status["UNKNOWN"] = 2] = "UNKNOWN";
})(Status = exports.Status || (exports.Status = {}));
/**
 * Error thrown when the Cerbos policy decision point server returns an unsuccessful response.
 *
 * @public
 */
class NotOK extends Error {
    constructor(
    /**
     * The status code returned by the Cerbos policy decision point server.
     */
    code, 
    /**
     * Additional error details.
     */
    details) {
        super(`gRPC error ${code} (${Status[code] ?? "unrecognized"}): ${details}`);
        this.code = code;
        this.details = details;
        this.name = this.constructor.name;
        Error.captureStackTrace(this, this.constructor);
    }
}
exports.NotOK = NotOK;
/**
 * Error thrown when input fails schema validation, if the {@link @cerbos/core#Client} is configured with {@link @cerbos/core#Options.onValidationError | onValidationError} set to `"throw"`.
 *
 * @public
 */
class ValidationFailed extends Error {
    constructor(
    /**
     * The validation errors that occurred.
     */
    validationErrors) {
        super("Input failed schema validation");
        this.validationErrors = validationErrors;
        this.name = this.constructor.name;
        Error.captureStackTrace(this, this.constructor);
    }
}
exports.ValidationFailed = ValidationFailed;
