// @cerbos/core/lib/protobuf/google/api/expr/v1alpha1/checked.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type_WellKnownType = exports.Type_PrimitiveType = exports.protobufPackage = void 0;
exports.protobufPackage = "google.api.expr.v1alpha1";
/** CEL primitive types. */
var Type_PrimitiveType;
(function (Type_PrimitiveType) {
    /** PRIMITIVE_TYPE_UNSPECIFIED - Unspecified type. */
    Type_PrimitiveType[Type_PrimitiveType["PRIMITIVE_TYPE_UNSPECIFIED"] = 0] = "PRIMITIVE_TYPE_UNSPECIFIED";
    /** BOOL - Boolean type. */
    Type_PrimitiveType[Type_PrimitiveType["BOOL"] = 1] = "BOOL";
    /**
     * INT64 - Int64 type.
     *
     * Proto-based integer values are widened to int64.
     */
    Type_PrimitiveType[Type_PrimitiveType["INT64"] = 2] = "INT64";
    /**
     * UINT64 - Uint64 type.
     *
     * Proto-based unsigned integer values are widened to uint64.
     */
    Type_PrimitiveType[Type_PrimitiveType["UINT64"] = 3] = "UINT64";
    /**
     * DOUBLE - Double type.
     *
     * Proto-based float values are widened to double values.
     */
    Type_PrimitiveType[Type_PrimitiveType["DOUBLE"] = 4] = "DOUBLE";
    /** STRING - String type. */
    Type_PrimitiveType[Type_PrimitiveType["STRING"] = 5] = "STRING";
    /** BYTES - Bytes type. */
    Type_PrimitiveType[Type_PrimitiveType["BYTES"] = 6] = "BYTES";
})(Type_PrimitiveType = exports.Type_PrimitiveType || (exports.Type_PrimitiveType = {}));
/** Well-known protobuf types treated with first-class support in CEL. */
var Type_WellKnownType;
(function (Type_WellKnownType) {
    /** WELL_KNOWN_TYPE_UNSPECIFIED - Unspecified type. */
    Type_WellKnownType[Type_WellKnownType["WELL_KNOWN_TYPE_UNSPECIFIED"] = 0] = "WELL_KNOWN_TYPE_UNSPECIFIED";
    /**
     * ANY - Well-known protobuf.Any type.
     *
     * Any types are a polymorphic message type. During type-checking they are
     * treated like `DYN` types, but at runtime they are resolved to a specific
     * message type specified at evaluation time.
     */
    Type_WellKnownType[Type_WellKnownType["ANY"] = 1] = "ANY";
    /** TIMESTAMP - Well-known protobuf.Timestamp type, internally referenced as `timestamp`. */
    Type_WellKnownType[Type_WellKnownType["TIMESTAMP"] = 2] = "TIMESTAMP";
    /** DURATION - Well-known protobuf.Duration type, internally referenced as `duration`. */
    Type_WellKnownType[Type_WellKnownType["DURATION"] = 3] = "DURATION";
})(Type_WellKnownType = exports.Type_WellKnownType || (exports.Type_WellKnownType = {}));
