// @cerbos/core/lib/protobuf/cerbos/engine/v1/engine.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trace_Event_Status = exports.Trace_Component_Kind = exports.PlanResourcesFilter_Kind = exports.PlanResourcesAst_LogicalOperation_Operator = exports.protobufPackage = void 0;
exports.protobufPackage = "cerbos.engine.v1";
var PlanResourcesAst_LogicalOperation_Operator;
(function (PlanResourcesAst_LogicalOperation_Operator) {
    PlanResourcesAst_LogicalOperation_Operator[PlanResourcesAst_LogicalOperation_Operator["OPERATOR_UNSPECIFIED"] = 0] = "OPERATOR_UNSPECIFIED";
    PlanResourcesAst_LogicalOperation_Operator[PlanResourcesAst_LogicalOperation_Operator["OPERATOR_AND"] = 1] = "OPERATOR_AND";
    PlanResourcesAst_LogicalOperation_Operator[PlanResourcesAst_LogicalOperation_Operator["OPERATOR_OR"] = 2] = "OPERATOR_OR";
    PlanResourcesAst_LogicalOperation_Operator[PlanResourcesAst_LogicalOperation_Operator["OPERATOR_NOT"] = 3] = "OPERATOR_NOT";
})(PlanResourcesAst_LogicalOperation_Operator = exports.PlanResourcesAst_LogicalOperation_Operator || (exports.PlanResourcesAst_LogicalOperation_Operator = {}));
var PlanResourcesFilter_Kind;
(function (PlanResourcesFilter_Kind) {
    PlanResourcesFilter_Kind[PlanResourcesFilter_Kind["KIND_UNSPECIFIED"] = 0] = "KIND_UNSPECIFIED";
    PlanResourcesFilter_Kind[PlanResourcesFilter_Kind["KIND_ALWAYS_ALLOWED"] = 1] = "KIND_ALWAYS_ALLOWED";
    PlanResourcesFilter_Kind[PlanResourcesFilter_Kind["KIND_ALWAYS_DENIED"] = 2] = "KIND_ALWAYS_DENIED";
    PlanResourcesFilter_Kind[PlanResourcesFilter_Kind["KIND_CONDITIONAL"] = 3] = "KIND_CONDITIONAL";
})(PlanResourcesFilter_Kind = exports.PlanResourcesFilter_Kind || (exports.PlanResourcesFilter_Kind = {}));
var Trace_Component_Kind;
(function (Trace_Component_Kind) {
    Trace_Component_Kind[Trace_Component_Kind["KIND_UNSPECIFIED"] = 0] = "KIND_UNSPECIFIED";
    Trace_Component_Kind[Trace_Component_Kind["KIND_ACTION"] = 1] = "KIND_ACTION";
    Trace_Component_Kind[Trace_Component_Kind["KIND_CONDITION_ALL"] = 2] = "KIND_CONDITION_ALL";
    Trace_Component_Kind[Trace_Component_Kind["KIND_CONDITION_ANY"] = 3] = "KIND_CONDITION_ANY";
    Trace_Component_Kind[Trace_Component_Kind["KIND_CONDITION_NONE"] = 4] = "KIND_CONDITION_NONE";
    Trace_Component_Kind[Trace_Component_Kind["KIND_CONDITION"] = 5] = "KIND_CONDITION";
    Trace_Component_Kind[Trace_Component_Kind["KIND_DERIVED_ROLE"] = 6] = "KIND_DERIVED_ROLE";
    Trace_Component_Kind[Trace_Component_Kind["KIND_EXPR"] = 7] = "KIND_EXPR";
    Trace_Component_Kind[Trace_Component_Kind["KIND_POLICY"] = 8] = "KIND_POLICY";
    Trace_Component_Kind[Trace_Component_Kind["KIND_RESOURCE"] = 9] = "KIND_RESOURCE";
    Trace_Component_Kind[Trace_Component_Kind["KIND_RULE"] = 10] = "KIND_RULE";
    Trace_Component_Kind[Trace_Component_Kind["KIND_SCOPE"] = 11] = "KIND_SCOPE";
    Trace_Component_Kind[Trace_Component_Kind["KIND_VARIABLE"] = 12] = "KIND_VARIABLE";
    Trace_Component_Kind[Trace_Component_Kind["KIND_VARIABLES"] = 13] = "KIND_VARIABLES";
})(Trace_Component_Kind = exports.Trace_Component_Kind || (exports.Trace_Component_Kind = {}));
var Trace_Event_Status;
(function (Trace_Event_Status) {
    Trace_Event_Status[Trace_Event_Status["STATUS_UNSPECIFIED"] = 0] = "STATUS_UNSPECIFIED";
    Trace_Event_Status[Trace_Event_Status["STATUS_ACTIVATED"] = 1] = "STATUS_ACTIVATED";
    Trace_Event_Status[Trace_Event_Status["STATUS_SKIPPED"] = 2] = "STATUS_SKIPPED";
})(Trace_Event_Status = exports.Trace_Event_Status || (exports.Trace_Event_Status = {}));
